docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
-# Manifest factorization through `.flux.yaml` configuration files
+# Manifest generation through `.flux.yaml` configuration files
 
-## Enabling search of `.flux.yaml` files
+This feature lets you generate Kubernetes manifests with a program,
+instead of having to include them in your git repo as YAML files. For
+example, you can use `kustomize` to patch a common set of resources to
+suit a particular environment.
 
-To enable it supply the command-line flag `--manifest-generation=true` to `fluxd`.
+## Setting manifest generation up
 
-## Goal
+Manifest generation is controlled by the flags given to `fluxd`, and
+`.flux.yaml` files in your git repo.
 
-It is a common pattern to run very similar resources in separate clusters. There are various scenarios in which this is
-required, the two main ones being:
+To enable it, pass the command-line flag `--manifest-generation=true`
+to `fluxd`.
 
-* Having a staging/canary cluster and production cluster. The resources from the staging cluster are regularly promoted
-  to the production cluster. In addition, there are long-term differences between the cluster resources (e.g. different
-  security keys, different database endpoints etc ...).
-* Federation. Different clusters run in separate regions, usually with very similar resources but different
-  configurations.
+For each path supplied with the flag `--git-path` (or `.` if none are
```

Comment:
Maybe we should introduce what `--git-path` is beforehand.

Reply:
Also, it may be worth clarifying that `.` means the root directory of the repository.

Reply:
And, I am surely biased but, wouldn't it make sense to (at least briefly) present a `.flux.yaml` file before getting into their interaction with `--git-path`?

As a reader I am not sure I would be able to understand what is going on without having a rough high-level idea of what the configuration will do on that `--git-path`

Reply:
> And, I am surely biased but, wouldn't it make sense to (at least briefly) present a `.flux.yaml` file before getting into their interaction with `--git-path`?

I think it would make sense to show it and indicate it will be explained later.

Reply:
I agree that explaining what `--git-path` does would make the docs better.

Reply:
Cool, OK. I was self-consciously not piling up too much text before getting to the good stuff, but you are both right that `--git-path` deserves a bit of explaining.

Reply:
I put a bit of explanation up-front, and substituted the term "target path" for `--git-path` afterwards. I got the term from the earlier text, and it seems OK to me, but I'm happy to adopt another.